api/src/license_key.rs

```diff
@@ -0,0 +1,108 @@
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum LicenseKeyStatus {
+    Valid,
+    Invalid,
+    Blacklisted,
+}
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct LicenseKeyProperties {
+    pub key_size: usize,
+    pub payload_size: usize,
+    pub checksum_size: usize,
+}
+
+impl LicenseKeyProperties {
+    pub fn default() -> LicenseKeyProperties {
+        LicenseKeyProperties {
+            key_size: 0,
+            payload_size: 0,
+            checksum_size: 0,
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct LicenseKey {
+    pub key: Vec<u8>,
+    pub payload: Vec<u8>,
+    pub checksum: Vec<u8>,
```

Comment:
It's u32, why bother to convert it to `Vec<u8>`

Reply:
You mean checksum? Not really, because checksum can be user defined operation so... i can be 64 bit or even 128 bit size or whatever user defined that.

But at that time you wrote this review it was unfinished behavior so you couldn't know it. Sorry...